src/inmanta/data/dataview.py

```diff
@@ -540,58 +540,86 @@ def get_extra_url_parameters(self) -> dict[str, str]:
+                        rps.environment,
+                        {const.SQL_RESOURCE_STATUS_SELECTOR} AS status
+                    FROM resource_persistent_state AS rps
+                    INNER JOIN resource AS r
+                        ON r.environment=rps.environment
+                        AND r.resource_id=rps.resource_id
+                    INNER JOIN resource_set_configuration_model AS rscm
+                        ON r.environment=rscm.environment
+                        AND r.resource_set_id=rscm.resource_set_id
+                        AND rscm.model=(SELECT version FROM latest_version)
+                    WHERE rps.environment=$1 AND NOT rps.is_orphan
+                )
+        """
+            if self.drop_orphans
+            else f"""
+                WITH latest_version AS (
+                    SELECT MAX(public.configurationmodel.version) as version
+                    FROM public.configurationmodel
+                    WHERE public.configurationmodel.released AND environment=$1
                 ), versioned_resource_state AS (
                     SELECT
                         rps.*,
                         CASE
                             -- try the cheap, trivial option first because the lookup has a big performance impact
-                            WHEN EXISTS (
-                                SELECT 1
-                                FROM resource AS r
-                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id AND r.model = (
-                                    SELECT version FROM latest_version
-                                )
-                            ) THEN (SELECT version FROM latest_version)
+                            WHEN NOT rps.is_orphan
+                                THEN (SELECT version FROM latest_version)
                             -- only if the resource does not exist in the latest released version, search for the latest
                             -- version it does exist in
                             ELSE (
-                                SELECT MAX(r.model)
+                                SELECT MAX(rscm.model)
                                 FROM resource AS r
-                                JOIN configurationmodel AS m
-                                    ON r.environment = m.environment AND r.model = m.version AND m.released = TRUE
-                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id
+                                INNER JOIN resource_set_configuration_model AS rscm
+                                    ON r.environment=rscm.environment AND r.resource_set_id=rscm.resource_set_id
+                                INNER JOIN configurationmodel AS m
+                                    ON rscm.environment=m.environment AND rscm.model=m.version
+                                WHERE r.environment=rps.environment AND r.resource_id=rps.resource_id AND m.released
```

Comment:
I expect this will kill the performance of the FE when showing orphans. 

Can we determine if it will? 

I wonder if we should simply write the version in which it was orphaned to the RPS table?